client/app/models/widget.js

```diff
@@ -0,0 +1,31 @@
+import DS from 'ember-data';
+import attr from 'ember-data/attr';
+import { belongsTo } from 'ember-data/relationships';
+
+export default DS.Model.extend({
+    name: attr('string', {defaultValue:'Unnamed Widget'}),
+    author: attr('string'),
+    width: attr('number'),
+    height: attr('number'),
+    query: attr({
```

Comment:
I think query needs to be just passed through as is. If a different object is stored in this field right now, it probably won't validate. Query here would be the whole data that gets posted. Maybe its better to rename this to request_data or something that is clearer.